src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -86,20 +86,27 @@ public async Task<Stream> GetGcDump(int pid, CancellationToken cancellationToken
             return stream;
         }
 
-        public async Task<IStreamWithCleanup> StartCpuTrace(int pid, TimeSpan duration, CancellationToken cancellationToken)
+        public async Task<IStreamWithCleanup> StartTrace(int pid, TraceProfile profile, TimeSpan duration, CancellationToken token)
         {
-            DiagnosticsMonitor monitor = new DiagnosticsMonitor(new CpuProfileConfiguration());
-            Stream stream = await monitor.ProcessEvents(pid, duration, cancellationToken);
-
-            return new StreamWithCleanup(monitor, stream);
-        }
+            IList<MonitoringSourceConfiguration> configurations = new List<MonitoringSourceConfiguration>();
+            if (profile.HasFlag(TraceProfile.Cpu))
+            {
+                configurations.Add(new CpuProfileConfiguration());
+            }
+            if (profile.HasFlag(TraceProfile.Http))
+            {
+                configurations.Add(new HttpRequestSourceConfiguration());
+            }
+            if (profile.HasFlag(TraceProfile.Logs))
+            {
+                configurations.Add(new LoggingSourceConfiguration());
+            }
+            if (profile.HasFlag(TraceProfile.Metrics))
+            {
+                configurations.Add(new MetricSourceConfiguration(5));
```

Comment:
The metrics interval is 60 seconds by default, which is longer than the default trace collection of 30 seconds. So you don't get any metrics data if using a collection time less than 60 seconds. I've temporarily set the interval to 5 seconds here, but we might need to come up with some other reasonable default and probably allow the caller to specify the interval.

Reply:
Can you propagate the configuration? I need to tweak this to 30 seconds anyway for prometheus.

Reply:
“Can you propagate the configuration?” I’m not sure what you mean by this.

I was looking a what dotnet-counters was using, and it has a default interval of 1 second. So I’m thinking of setting the default to the same and then adding a query parameter to allow that to be overridden.

Reply:
👍